src/components/SearchAndFilter.js

```diff
@@ -47,11 +47,17 @@ class SearchAndFilter extends React.Component {
       <React.Fragment key={length}>
         <label>Ingredient: </label>
         <input type="text" value={ingredients[length]} onChange={(event) => this.setIngredient(length, event)}/>
-        <Button onClick={this.addInput} type="button">+</Button>
+        <button id={`form-add-${length}`} onClick={this.addInput} type="button">+</button>
         <br></br>
       </React.Fragment>
     )
-    this.setState({inputFields: inputs});
+    this.setState({inputFields: inputs}, () => {
+      if (length != 0) {
+        document.getElementById(`form-add-${length-1}`).style.visibility = "hidden";
```

Comment:
nit: sometimes it's better to have a class in your CSS that has visibility:hidden, like .hidden {...} and do document.getElementById(...).classList.toggle('visible') or .classList.add/remove

https://developer.mozilla.org/en-US/docs/Web/API/Element/classList